src/main/java/com/st/utopia/agent/controller/SeatTickets.java

```diff
@@ -0,0 +1,189 @@
+package com.st.utopia.agent.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.RestTemplate;
+
+import com.st.utopia.agent.model.PaymentAmount;
+import com.st.utopia.agent.model.SeatLocation;
+import com.st.utopia.agent.model.Ticket;
+import com.st.utopia.agent.model.User;;
+
+/**
+ * Controller for agents using this user-facing interface
+ *
+ * @author Jonathan Lovelace
+ * @author Salem O
+ */
+@RestController
+public class SeatTickets {
+	@Autowired
+	private RestTemplate restTemplate;
+
+	@Value("${utopia.search.API}")
+	private String searchAPI;
+
+	@Value("${utopia.booking.API}")
+	private String bookingAPI;
+
+	@Value("${utopia.cancellation.API}")
+	private String cancellationAPI;
+	
+	/**
+	 * Helper method to reduce the amount of repetitive code required for "get-all"
+	 * methods.
+	 * @param url the URL to send the REST request to
+	 * @param <T> the type we expect
+	 * @return the response the server sent
+	 */
+	private <T> ResponseEntity<T> methodCall(final String url, final HttpMethod method) {
+		return restTemplate.exchange(url, method, null,
```

Comment:
To make the difference between this and the following method clearer, this method could simply call it with `null` for the final parameter.

Reply:
However, it would be a little too complicated to fix this in the short time that we have